lib/Cake/Network/CakeSocket.php

```diff
@@ -116,6 +117,32 @@ class CakeSocket {
  */
 	public function __construct($config = array()) {
 		$this->config = array_merge($this->_baseConfig, $config);
+
+		// These TLS versions are not supported by older PHP versions,
+		// so we have to conditionally set them if they are supported.
+		$conditionalCrypto = array(
+			'tlsv1_1_client' => 'STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT',
```

Comment:
If we set these TLS versions depending on the PHP version, shouldn't we test the list of them conditionally?

Reply:
Perfect! Works good.